### PR TITLE
[ExportVerilog] Add explicit bitcast for sub expressions 

### PR DESCRIPTION
**Pull request by uenoku**
Unfortunately spyglass infers bitwidth of sub expression incorrectly even when they are not toplevel. This PR will surpass the lint warnings.  
This revert https://github.com/llvm/circt/pull/3094 and adds explicit bitcast for every sub expression when `explicitBitcastAddMul` option is true.

### CODE REVIEW

**Review by mikeurbach — APPROVED** [review]
I think it makes sense... *shakes fist at spyglass*

File: include/circt/Support/LoweringOptions.h [hunk 1]
```diff
@@ -97,7 +97,6 @@ struct LoweringOptions {
   unsigned emittedLineLength = DEFAULT_LINE_LENGTH;
 
   /// Add an explicit bitcast for avoiding bitwidth mismatch LINT errors.
-  /// TODO: Change the name of option since it is not limit to add/mul anymore.
   bool explicitBitcastAddMul = false;
```
[the remark this answers is not in the document]
Comment (uenoku) [reply include/circt/Support/LoweringOptions.h]: Yeah, this was my old TODO..:) I changed the name this time.

[the remark this answers is not in the document]
Comment (mikeurbach) [reply include/circt/Support/LoweringOptions.h]: Thanks!

**Review by youngar — APPROVED** [review]
LGTM

**Pull request merged**